src/newgrf_gui.cpp

```diff
@@ -1314,10 +1320,29 @@ struct NewGRFWindow : public Window, NewGRFScanCallback {
 		this->SetWidgetDisabledState(WID_NS_PRESET_SAVE, has_missing);
 	}
 
-	EventState OnKeyPress(WChar key, uint16 keycode) override
+	EventState OnKeyPressForActiveList(uint16 keycode)
 	{
-		if (!this->editable) return ES_NOT_HANDLED;
+		/* Get total actives and index of current selection. */
+		int active_pos = 0;
+		int active_count = 0;
+		for (GRFConfig *c = this->actives; c != nullptr; c = c->next, active_count++) {}
+		for (GRFConfig *c = this->actives; c != nullptr && c != this->active_sel; c = c->next, active_pos++) {}
 
+		if (this->vscroll2->UpdateListPositionOnKeyPress(active_pos, keycode) == ES_NOT_HANDLED) return ES_NOT_HANDLED;
+
+		active_pos = std::min(active_pos, active_count);
+
+		if (this->actives != nullptr) {
+			this->SelectActive(active_pos);
+			this->vscroll->ScrollTowards(active_pos == active_count - 1 ? active_pos + 1 : active_pos);
```

Comment:
I would not deviate the behavior here from all the other behavior, so passing only `active_pos` should be fine as well. I find it even odd that it jumps two places for the last. I would for example expect that a page-down shows me the next N elements, but with this logic you can completely miss an element.
If I got 13 NewGRFs in the list and start with the minimum window size at the top, I see 1-6. After page down I see 2-7 and after another page down I see. 9-13(+1). In other words, I completely missed the NewGRF at line 8.
Removing this also removes the need to count the number of elements, and even those should already be known and stored in `this->vscroll->GetCount()`.